File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationDepartEvent.java
package com.mapbox.services.android.navigation.v5.navigation.metrics;

import android.annotation.SuppressLint;
import android.support.annotation.Keep;

@Keep
@SuppressLint("ParcelCreator")
class NavigationDepartEvent extends NavigationEvent {
  private static final String NAVIGATION_DEPART = "navigation.depart";

  NavigationDepartEvent(PhoneState phoneState) {
    super(phoneState);
  }

  @Override
  protected String getEventName() {
    return NAVIGATION_DEPART;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/NavigationDepartEvent.kt
package com.mapbox.services.android.navigation.v5.internal.navigation.metrics

import android.annotation.SuppressLint
import com.mapbox.services.android.navigation.v5.navigation.metrics.NavigationMetrics

@SuppressLint("ParcelCreator")
internal class NavigationDepartEvent(
    phoneState: PhoneState
) : NavigationEvent(phoneState) {

    override fun getEventName(): String = NavigationMetrics.DEPART
}
